### PR TITLE
[FIX_BUG][*] fix some bugs from sonar analysis

### PR DESCRIPTION
**Pull request by CalvinKirs**
## What is the purpose of the pull request

fix some bugs from sonar analysis
#3321 
## Verify this pull request

This pull request is already covered by existing tests, such as *(please describe tests)*.

### CODE REVIEW

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/zk/ZookeeperCachedOperator.java [hunk 1]
```diff
@@ -83,6 +83,7 @@ public void close() {
         try {
             Thread.sleep(500);
         } catch (InterruptedException ignore) {
+            Thread.currentThread().interrupt();
```
Comment (gabrywu) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/zk/ZookeeperCachedOperator.java:86]: why not simply remove the catch block ?

**Comment by gabrywu** [comment]
referenced blog [here](https://www.cnblogs.com/onlywujun/p/3565082.html)

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/zk/ZookeeperCachedOperator.java (hunk 1 above)
Comment (CalvinKirs) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/zk/ZookeeperCachedOperator.java:86]: > why not simply remove the catch block ?

Thank you very much for your suggestion, I will finish it later

**Comment by CalvinKirs** [comment]
> referenced blog [here](https://www.cnblogs.com/onlywujun/p/3565082.html) about interrupt

Hello, I tried to do what you said, but I found that there are too many places to call this method, so the changes will be relatively large

File: dolphinscheduler-remote/src/main/java/org/apache/dolphinscheduler/remote/handler/NettyClientHandler.java [hunk 2]
```diff
@@ -75,7 +75,7 @@ public NettyClientHandler(NettyRemotingClient nettyRemotingClient, ExecutorServi
      * @throws Exception
      */
     @Override
-    public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+    public void channelInactive(ChannelHandlerContext ctx) {
```
Comment (gabrywu) [inline dolphinscheduler-remote/src/main/java/org/apache/dolphinscheduler/remote/handler/NettyClientHandler.java:78]: why remove 'throws Exception'?

File: dolphinscheduler-remote/src/main/java/org/apache/dolphinscheduler/remote/handler/NettyServerHandler.java [hunk 3]
```diff
@@ -139,7 +138,7 @@ public void run() {
      * @throws Exception
      */
     @Override
-    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
```
Comment (gabrywu) [inline dolphinscheduler-remote/src/main/java/org/apache/dolphinscheduler/remote/handler/NettyServerHandler.java:141]: why remove 'throws Exception'?

File: e2e/src/test/java/org/apache/dolphinscheduler/locator/project/ProcessInstanceLocator.java [hunk 4]
```diff
@@ -21,7 +21,7 @@
 public class ProcessInstanceLocator {
     // jump Process Instance page
     //click Process Instance name
-    public static final By CLICK_PROCESS_INSTANCE_NAME = By.xpath("//div[3]/div/ul/li[2]");
+    public static final By CLICK_PROCESS_INSTANCE_NAME = By.xpath("//div[4]/div/ul/li[2]");
```
Comment (gabrywu) [inline e2e/src/test/java/org/apache/dolphinscheduler/locator/project/ProcessInstanceLocator.java:24]: Is this a bug?Is there a related issue?

**Comment by CalvinKirs** [comment]
Exception 'java.lang.Exception' is never thrown in the method 




On 07/31/2020 14:57，gabry.wu<notifications@github.com> wrote：

@gabrywu requested changes on this pull request.

In dolphinscheduler-remote/src/main/java/org/apache/dolphinscheduler/remote/handler/NettyClientHandler.java:

> @@ -75,7 +75,7 @@ public NettyClientHandler(NettyRemotingClient nettyRemotingClient, ExecutorServi
      * @throws Exception
      */
     @Override
-    public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+    public void channelInactive(ChannelHandlerContext ctx) {


why remove 'throws Exception'?

In dolphinscheduler-remote/src/main/java/org/apache/dolphinscheduler/remote/handler/NettyServerHandler.java:

> @@ -139,7 +138,7 @@ public void run() {
      * @throws Exception
      */
     @Override
-    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {


why remove 'throws Exception'?

In e2e/src/test/java/org/apache/dolphinscheduler/locator/project/ProcessInstanceLocator.java:

> @@ -21,7 +21,7 @@
 public class ProcessInstanceLocator {
     // jump Process Instance page
     //click Process Instance name
-    public static final By CLICK_PROCESS_INSTANCE_NAME = By.xpath("//div[3]/div/ul/li[2]");
+    public static final By CLICK_PROCESS_INSTANCE_NAME = By.xpath("//div[4]/div/ul/li[2]");


Is this a bug?Is there a related issue?

File: e2e/src/test/java/org/apache/dolphinscheduler/locator/project/ProcessInstanceLocator.java (hunk 4 above)
Comment (CalvinKirs) [reply e2e/src/test/java/org/apache/dolphinscheduler/locator/project/ProcessInstanceLocator.java:24]: This is to solve the bug of e2e test

File: dolphinscheduler-remote/src/main/java/org/apache/dolphinscheduler/remote/handler/NettyServerHandler.java (hunk 3 above)
Comment (CalvinKirs) [reply dolphinscheduler-remote/src/main/java/org/apache/dolphinscheduler/remote/handler/NettyServerHandler.java:141]: Exception 'java.lang.Exception' is never thrown in the method

Comment (CalvinKirs) [reply dolphinscheduler-remote/src/main/java/org/apache/dolphinscheduler/remote/handler/NettyServerHandler.java:141]: If it occurs, it should also be handed over to the upper caller.

File: e2e/src/test/java/org/apache/dolphinscheduler/locator/project/ProcessInstanceLocator.java (hunk 4 above)
Comment (gabrywu) [reply e2e/src/test/java/org/apache/dolphinscheduler/locator/project/ProcessInstanceLocator.java:24]: OK,cool

**Comment by gabrywu** [comment]
Let's invite @qiaozhanwei @dailidong  to review this PR, if there're no opposite suggestions, PR can be merged

**Pull request merged**